### ISSUE TITLE
imread during inference load the image as uint8

### ISSUE THREAD

**Issue by ZhenghaoFei** [ClementPinard/SfmLearner-Pytorch#142]
Hi Clement,

Thank you for your code and nice implementation! 

I catched a small bug here. Using the downloaded pretrained model to run run_inference.py on resized (416*128) test data, I got bad disparity results. 

![image](https://user-images.githubusercontent.com/11188569/207212990-7a8f82c2-17ac-4185-9117-bf8b3f634e37.png)

Running inference of the same model on original image (full res) was totally fine. 

The bug is clearly related to the imread() on line 58 and resize() on line 62 [https://github.com/ClementPinard/SfmLearner-Pytorch/blob/master/run_inference.py](url), the imread() loads image as uint8 (0-255) and the resize will normalize the image pixel value into (0-1), thats why  using the full res image was fine. But when the images are already resized, the resize() function will be skiped and uint8 image is going to feed in the net and get bad results.

I added  img = img/255.0 after imread and fixed the problem.

![image](https://user-images.githubusercontent.com/11188569/207213917-5f83ea36-3264-4350-a6ff-c7269f9064f3.png)

I am not sure why this bug did not affect you and others (might due to my system environment?, python 3.88).

Best,
Zhenghao

**Comment by ZhenghaoFei** [comment]
Related to #136 and #125

**Comment by ClementPinard** [comment]
Hello, you are right, there have been multiple issues related to image loading, resizing and normalizing for this script.

I think you are right in your assumption. Actually, the docs itself says that resize normalizes the image with skimaghe.util.img_as_float before actually resizing.

A very quick fix is then not to divide by 255, but simply call this function before anything else when the image is loaded.

I will push a fix for that, thanks for finding the bug that has been here for multiple years !

**Comment by ClementPinard** [comment]
If you confirm the fix works for you, I'll merge it and the bug will be fixed.

Thanks for you help !
